### PR TITLE
[MAGETWO-85227]:+ This commit fixes a incorrect fetch of column value…

### PR DESCRIPTION
**Pull request by jspoe**
[MAGETWO-85227]:+ This commit fixes a incorrect fetch of column values on a loaded collection

### Description
Clones a unloaded collection for the grid to use.
Resets the collection limit to remove the limit of the collection

### Fixed Issues (if relevant)
1. magento/magento2#12594: Couldn't select all items on Review grid

### Manual testing scenarios
1. login Magento admin
2. from main menu, select MARKETING -> User Content -> Reviews
3. select Select All under Actions checkbox

**Expected Result**
All review items are selected.

1. login Magento admin
2. from main menu, select MARKETING -> User Content -> Reviews
3. select Select all visible under Actions checkbox

**Expected Result**
Only visible review items are selected.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @jspoe. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18060) <br/>All committers have signed the CLA.

File: app/code/Magento/Backend/Block/Widget/Grid/Extended.php
Comment (slavvka) [inline app/code/Magento/Backend/Block/Widget/Grid/Extended.php:1268]: It's a breaking change. Please look into the guide https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/

Comment (jspoe) [reply app/code/Magento/Backend/Block/Widget/Grid/Extended.php:1268]: Hi @slavvka 

See my comment in https://github.com/magento/magento2/issues/12594#issuecomment-421357028 
How do you see a solution for this change? Because both won't be good enough, I think.

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.
Thanks for collaboration!

**Comment by slavvka** [comment]
Hey @jspoe  Anyway this change should be applied to the latest code. Please reopen it to 2.4

**Pull request closed without merging**
